src/services/mongoose/wallet-invoices.ts

```diff
@@ -161,3 +139,11 @@ export const WalletInvoicesRepository = (): IWalletInvoicesRepository => {
     deleteUnpaidOlderThan,
   }
 }
+
+const walletInvoiceFromRaw = (result): WalletInvoice => ({
+  paymentHash: result.id as PaymentHash,
```

Comment:
I think this should be `_id` right? The schema doesn't define an `id` field:
```
const invoiceUserSchema = new Schema({
  _id: String, // hash of invoice
  walletId: String,

  // usd equivalent. sats is attached in the invoice directly.
  // optional, as BTC wallet doesn't have to set a sat amount when creating the invoice
  usd: Number,

  timestamp: {
    type: Date,
    default: Date.now,
  },

  selfGenerated: {
    type: Boolean,
    default: true,
  },

  pubkey: {
    type: String,
    require: true,
  },

  paid: {
    type: Boolean,
    default: false,
  },
})

```
But there is some inconsistency going on as `_id` should be reserved for mongo `ObjectId`s but here its a `string`... hmm

Reply:
> I think this should be `_id` right? The schema doesn't define an `id` field:
> 
> ```
> const invoiceUserSchema = new Schema({
>   _id: String, // hash of invoice
>   walletId: String,
> 
>   // usd equivalent. sats is attached in the invoice directly.
>   // optional, as BTC wallet doesn't have to set a sat amount when creating the invoice
>   usd: Number,
> 
>   timestamp: {
>     type: Date,
>     default: Date.now,
>   },
> 
>   selfGenerated: {
>     type: Boolean,
>     default: true,
>   },
> 
>   pubkey: {
>     type: String,
>     require: true,
>   },
> 
>   paid: {
>     type: Boolean,
>     default: false,
>   },
> })
> ```
> 
> But there is some inconsistency going on as `_id` should be reserved for mongo `ObjectId`s but here its a `string`... hmm

it shouldn't be `_id` if we consider `_id` are `ObjectId`

Reply:
agree - it should be `id` if anything but that would require a migration or? My point is as the code currently stands:
```
paymentHash: result.id as PaymentHash,
```
shouldn't work because the schema doesn't define `id` so I guess its using mongooses virtual field.